app/src/main/java/de/xikolo/controllers/main/CertificateListAdapter.kt

```diff
@@ -0,0 +1,134 @@
+package de.xikolo.controllers.main
+
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.LinearLayout
+import android.widget.TextView
+import butterknife.BindView
+import butterknife.ButterKnife
+import de.xikolo.App
+import de.xikolo.R
+import de.xikolo.config.GlideApp
+import de.xikolo.controllers.helper.DownloadViewHelper
+import de.xikolo.models.Course
+import de.xikolo.models.DownloadAsset
+import de.xikolo.models.Enrollment
+import java.util.*
+
+class CertificateListAdapter(private val fragment: CertificateListFragment, private val callback: OnCertificateCardClickListener) : RecyclerView.Adapter<CertificateListAdapter.CertificateViewHolder>() {
+
+    companion object {
+        val TAG = CertificateListAdapter::class.java.simpleName!!
+    }
+
+    private var courseList: MutableList<Course> = ArrayList()
+
+    fun update(courseList: MutableList<Course>) {
+        this.courseList = courseList
+        notifyDataSetChanged()
+    }
+
+    fun clear() {
+        courseList.clear()
+        this.notifyDataSetChanged()
+    }
+
+    override fun getItemCount(): Int {
+        return courseList.size
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CertificateViewHolder {
+        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_certificate_list, parent, false)
+        return CertificateViewHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: CertificateViewHolder, position: Int) {
+        val course = courseList[position]
+
+        holder.textTitle.text = course.title
+
+        holder.header.setOnClickListener { _ -> callback.onCourseClicked(course.id) }
+
+        if (course.imageUrl != null)
+            GlideApp.with(App.getInstance()).load(course.imageUrl).into(holder.courseImage)
+        else
+            holder.courseImage.visibility = View.GONE
+
+        holder.container.removeAllViews()
+
+        val enrollment: Enrollment? = Enrollment.getForCourse(course.id)
+        if (enrollment != null) {
```

Comment:
These two lines can be simplified: `Enrollment.getForCourse(course.id)?.let { enrollment }`